datafusion/functions-nested/src/repeat.rs

```diff
@@ -124,19 +125,47 @@ impl ScalarUDFImpl for ArrayRepeat {
         &self.aliases
     }
 
+    fn coerce_types(&self, arg_types: &[DataType]) -> Result<Vec<DataType>> {
+        if arg_types.len() != 2 {
+            return exec_err!("array_repeat expects two arguments");
+        }
+
+        let element_type = &arg_types[0];
+        let first = element_type.clone();
+
+        let count_type = &arg_types[1];
+
+        // Coerce the second argument to Int64/UInt64 if it's a numeric type
+        let second = match count_type {
+            DataType::Int8 | DataType::Int16 | DataType::Int32 | DataType::Int64 => {
```

Comment:
@korowa @alamb 
converting negative numbers to UInt was giving: `arrow error, can't cast negative number to Uint`.

So, converted the Int values to `Int64` .

Then using `array_repeat` inner function to convert `Int64` to `UInt64` type

Reply:
Non-necessary suggestion: due to safe cast converting negative values to nulls, perhaps we should also add a simple test to verify that count array with multiple values with nulls will be processed as expected, like
```sql
select array_repeat('x', column1) from (values (-1), (2), (-3));
```